### ISSUE TITLE
[BUG] PCI Passthrough Entirely Broken for `RTL8111/8168/8411 PCI Express Gigabit Ethernet Controller` - `Realtek Semiconductor Co., Ltd.`

### ISSUE THREAD

**Issue by irishgordo** [harvester/harvester#5371]
**Describe the bug**
PCI Passthrough will enable for the device, the device claims get built, but a VM with those devices attached will never start.

**Pre-Reqs & Reproduce:**
1. Have a:
```
       description: Ethernet interface
       product: RTL8111/8168/8411 PCI Express Gigabit Ethernet Controller
       vendor: Realtek Semiconductor Co., Ltd.
       physical id: 0
       bus info: pci@0000:08:00.0
       logical name: enp8s0
       version: 15
       serial: 1c:86:0b:25:23:d7
       size: 1Gbit/s
       capacity: 1Gbit/s
       width: 64 bits
       clock: 33MHz
       capabilities: pm msi pciexpress msix bus_master cap_list ethernet physical tp mii 10bt 10bt-fd 100bt 100bt-fd 1000bt-fd autonegotiation
       configuration: autonegotiation=on broadcast=yes driver=r8169 driverversion=5.14.21-150400.24.92-default duplex=full firmware=rtl8168h-2_0.0.2 02/26/15 latency=0 link=yes multicast=yes port=twisted pair speed=1Gbit/s
       resources: irq:17 ioport:5000(size=256) memory:92e04000-92e04fff memory:92e00000-92e03fff

```
2. Enable PCI Passthrough for the devices
3. See it's enabled, add to a running vm

**Environment**
 - Harvester ISO version: v1.2.1 
 - Underlying Infrastructure: Bare Metal HP DL160

**Additional context**
Note that while PCI Passthrough fails for the `RTL8111/8168/8411 PCI Express Gigabit Ethernet Controller` by `Realtek Semiconductor Co., Ltd.` , the ability to have Networks -> Network Configs -> VM Networks on those ports works just fine.
It's just the PCI PassThrough portion.

**Comment by ibrokethecloud** [comment]
a support bundle when this issue can be reproduced would be appreciated. pcidevices-controller logs should indicate the reason for the same.

**Comment by irishgordo** [comment]
in v1.3.0, the VM does start, but the NIC is not present.
```

ubuntu@newtest:~$ ip a
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
2: enp1s0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP group default qlen 1000
    link/ether f2:2f:35:c5:72:57 brd ff:ff:ff:ff:ff:ff
    inet 192.168.104.241/24 metric 100 brd 192.168.104.255 scope global dynamic enp1s0
       valid_lft 86385sec preferred_lft 86385sec
    inet6 fe80::f02f:35ff:fec5:7257/64 scope link 
       valid_lft forever preferred_lft forever
ubuntu@newtest:~$ exit
logout
```
Based on:
![Screenshot from 2024-04-17 15-25-25](https://github.com/harvester/harvester/assets/5370752/b8ae6929-4954-4dfa-bac7-90cbb5310b56)

It's strange...
[supportbundle_295d3257-9675-46d6-aefe-14fafd2e20d9_2024-04-17T22-25-39Z.zip](https://github.com/harvester/harvester/files/15017685/supportbundle_295d3257-9675-46d6-aefe-14fafd2e20d9_2024-04-17T22-25-39Z.zip)

**Comment by irishgordo** [comment]
v1.2.1 support bundle with stuck vm realtek

[supportbundle_b1441488-b51a-4dcf-8728-ef2b8c12b55a_2024-04-18T01-47-55Z.zip](https://github.com/harvester/harvester/files/15018860/supportbundle_b1441488-b51a-4dcf-8728-ef2b8c12b55a_2024-04-18T01-47-55Z.zip)

**Comment by ibrokethecloud** [comment]
The device plugin name for the realtek nic exceeds 108 chars as a result the plugin is not initialised. 

v1.2.1 contains pcidevices v0.2.5 which does not have the fix for long device names.

This has been fixed in pcidevices v0.2.7 which is included in v1.2-head and will be made available in v1.2.2

**Comment by irishgordo** [comment]
:+1: @ibrokethecloud  thanks for this mention - seeing that it's fixed I'll go ahead and close this out :smile: :+1:
